packages/react-ds/src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from 'react'
+
+import { IPagination } from '@venice/core/models'
+import styles from '@venice/styles/components/Pagination.module.scss'
+
+import {
+  getRangeIndexes,
+  hasNextRange,
+  hasPreviousRange,
+} from './helpers/PaginationHelper'
+import AngleLeft from './icons/AngleLeft'
+import AngleRight from './icons/AngleRight'
+import PaginationRange from './PaginationRange'
+
+interface IPaginationProps extends IPagination {
+  /** Total pages to be paginated. */
+  count?: number
+  /** Current page, starting in zero. */
+  page?: number
+  /** Pagination status. When true, pagination buttons are disabled. */
+  isLoading?: boolean
+  /** Change page event. */
+  onChangePage?: (page: number) => void
+}
+
+const Pagination: React.FC<IPaginationProps> = ({
+  count = 0,
+  page = 0,
+  isLoading = false,
+  onChangePage = () => null,
+}: IPaginationProps) => {
+  const [isMobile, setIsMobile] = useState(false)
+
+  useEffect(() => {
+    let doit: NodeJS.Timeout
+    const timeOut = 200
+    const maxPhoneWidth = 420
+
+    const checkDevice = () =>
+      window.innerWidth <= maxPhoneWidth
+        ? setIsMobile(true)
+        : setIsMobile(false)
+
+    const onResize = () => {
+      clearTimeout(doit)
+      doit = setTimeout(() => checkDevice(), timeOut)
+    }
+    checkDevice()
+    window.addEventListener('resize', onResize)
```

Comment:
you need clear this listener on component unmout, otherwise, you can have multiple listeners running in same time. And you can use the same way to clear timeout.

```
useEffect(() => {
   const doit = timeout
   window.addListner(...)
  return () => {
     clearTimeout(doit)
    window.romoveListner(onResize)
  }
})
```